src/wazuh_db/wdb_global.c

```diff
@@ -1058,17 +1058,45 @@ wdbc_result wdb_global_sync_agent_info_get(wdb_t *wdb, int* last_agent_id, char
 }
 
 char* wdb_global_calculate_agent_group_csv(wdb_t *wdb, int id) {
-    cJSON* j_agent_groups = wdb_global_select_group_belong(wdb, id);
     char* result = NULL;
-    if (j_agent_groups) {
-        cJSON* j_group_name = NULL;
-        cJSON_ArrayForEach(j_group_name, j_agent_groups) {
-            wm_strcat(&result, cJSON_GetStringValue(j_group_name), MULTIGROUP_SEPARATOR);
+    sqlite3_stmt *stmt = NULL;
+
+    if (!wdb->transaction && wdb_begin2(wdb) < 0) {
+        mdebug1("Cannot begin transaction");
+        return NULL;
+    }
+
+    if (wdb_stmt_cache(wdb, WDB_STMT_GLOBAL_SELECT_GROUP_BELONG) < 0) {
+        mdebug1("Cannot cache statement");
+        return NULL;
+    }
+
+    stmt = wdb->stmt[WDB_STMT_GLOBAL_SELECT_GROUP_BELONG];
+
+    if (sqlite3_bind_int(stmt, 1, id) != SQLITE_OK) {
+        merror("DB(%s) sqlite3_bind_int(): %s", wdb->id, sqlite3_errmsg(wdb->db));
+        return NULL;
+    }
+
+    int _status = SQLITE_ROW;
+
+    while ((_status = wdb_step(stmt)) == SQLITE_ROW) {
+        char * group_hash = (char *) sqlite3_column_text(stmt, 0);
+
+        if (group_hash == NULL) {
+            mdebug1("Group hash is NULL");
+            continue;
         }
-        cJSON_Delete(j_agent_groups);
+
+        if (result != NULL && WDB_MAX_RESPONSE_SIZE < strlen(result) + strlen(group_hash) + 1) {
+            mdebug1("The agent's groups exceed the socket maximum response size.");
```

Comment:
In this case, I wouldn't return an incomplete group CSV string but fail completely.

Reply:
The same as above, I think the log messages is what we want for now, adding a pagination mechanism probably is out of scope.

Reply:
I agree about the error message, and the pagination isn't an option.
Just suggested to return.
But we don't have time.